File: app/src/main/java/im/fdx/v2ex/view/CustomChrome.java
package im.fdx.v2ex.view;

import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.Uri;
import android.support.customtabs.CustomTabsIntent;
import android.support.v4.content.ContextCompat;

import im.fdx.v2ex.MyReceiver;
import im.fdx.v2ex.R;

/**
 * Created by fdx on 2017/3/23.
 */
public class CustomChrome {

    private Context context;
    private final CustomTabsIntent.Builder builder;
    private final Intent sendIntent;

    public CustomChrome(Context context) {
        this.context = context;
        builder = new CustomTabsIntent.Builder();
        builder.setShowTitle(true);
        builder.setToolbarColor(ContextCompat.getColor(context, R.color.primary));

        Bitmap icon = BitmapFactory.decodeResource(context.getResources(), R.drawable.ic_share_white_24dp);
        sendIntent = new Intent(context, MyReceiver.class);

        PendingIntent pendingIntent = PendingIntent.getBroadcast(context, 0,
                sendIntent, PendingIntent.FLAG_UPDATE_CURRENT);
        builder.setActionButton(icon, "分享该页面", pendingIntent, true);
    }

    public void load(String url) {
        sendIntent.setData(Uri.parse(url));
        CustomTabsIntent customTabsIntent = builder.build();
        customTabsIntent.launchUrl(context, Uri.parse(url));
    }
}

File: app/src/main/java/im/fdx/v2ex/view/CustomChrome.kt
package im.fdx.v2ex.view

import android.content.ActivityNotFoundException
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.util.TypedValue
import androidx.browser.customtabs.CustomTabColorSchemeParams
import androidx.browser.customtabs.CustomTabsIntent
import androidx.core.content.ContextCompat
import androidx.core.content.ContextCompat.startActivity
import im.fdx.v2ex.R
import im.fdx.v2ex.utils.extensions.getColorFromAttr


/**
 * Created by fdx on 2017/3/23.
 * 无法设置文字颜色
 */
class CustomChrome(private val context: Context) {
    private val builder: CustomTabsIntent.Builder = CustomTabsIntent.Builder()

    init {
        val typedValue = TypedValue()
        val resolved = context.theme.resolveAttribute(R.attr.toolbar_background, typedValue, true)
        val colorBackground: Int
        if (resolved) {
            if (typedValue.resourceId != 0) {
                colorBackground = ContextCompat.getColor(context, typedValue.resourceId)
            } else {
                colorBackground = typedValue.data
            }
        } else {
            colorBackground = ContextCompat.getColor(context, android.R.color.white)
        }
        builder.setShowTitle(true)
        builder.setDefaultColorSchemeParams(
            CustomTabColorSchemeParams.Builder()
                .setToolbarColor(context.getColorFromAttr(R.attr.toolbar_background, R.color.chrome_tab)).build()
        )
        builder.setShareState(CustomTabsIntent.SHARE_STATE_ON)
    }

    fun load(url: String) {

        val customTabsIntent = builder.build()
        try {
            customTabsIntent.launchUrl(context, Uri.parse(url))
        } catch (e: ActivityNotFoundException) {
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
            context.startActivity(intent)
        }
    }
}
